Durability.md

```diff
@@ -0,0 +1,338 @@
+---
+RFC: 33
+Status: Draft
+---
+
+# Durability High Level Design Document
+
+## Problem Statement
+
+Valkey is an in-memory data store designed for high performance. Valkey currently uses an asynchronous replication model for high throughput and low latency, which means writes acknowledged to clients may be lost if replicas lag, crash, or are promoted while out-of-sync. Because all data resides in memory by default, process restarts or crashes risk losing recent writes. Although periodic snapshots (RDB) and append-only logs (AOF) provide persistence, they entail trade-offs: snapshots may omit recent writes and AOF only supports single-node durability. In cluster mode, data loss remains possible when a replica is promoted prematurely. Shifting durability responsibility to clients (via WAIT) is not intuitive and places burden on the application layer.
+
+## Motivation
+
+While Valkey’s asynchronous model prioritizes performance, many production workloads demand stronger guarantees where acknowledged writes will never be lost, even under failures. This specification introduces a unified durability layer that defines when a write is truly “safe” ensures deterministic behavior during failover, and simplifies client integration by removing the need for ad-hoc mechanisms like `WAIT`.
+
+To achieve this, the specification adopts a formally defined state machine replication model, namely [RAFT](https://en.wikipedia.org/wiki/Raft_(algorithm)), making the guarantees explicit and verifiable acting as a precise contract between the system and its users.
+
+## Requirements
+
+Please refer [https://github.com/valkey-io/valkey-rfc/pull/29/](https://github.com/valkey-io/valkey-rfc/pull/29/files) for detailed requirements
+
+* Support all the existing commands in Valkey.
+* Server level durability configuration - Valkey must allow enabling/disabling durability mode at startup.
+* Read Committed isolation level - Every write operation is serialized on primary and client won't see a volatile key.
+* Maintain current deployment architecture - No additional dependencies.
+* Support standalone mode.
```

Comment:
I feel like we overload what standalone means. Standalone is a single instance installation. Cluster was supposed to be multi-node configurations.